p2p/server.py

```diff
@@ -305,6 +359,10 @@ def __init__(self,
     async def do_handshake(self, peer: BasePeer) -> None:
         await peer.do_p2p_handshake(),
         await peer.do_sub_proto_handshake()
+        self._start_peer(peer)
+
+    def _start_peer(self, peer: BasePeer) -> None:
```

Comment:
It seems like we *should* be able to get rid of this by instead using a `PeerPool` class in the tests who's `start_peer` method was overridden.

Reply:
Indeed, but for that I'd need to change `get_server()` and `receiver_server_with_dumb_peer()` to use that mock `PeerPool`. I'd prefer to keep it as it is since I intend to rewrite the tests in `test_server.py`, and then I should be able to get rid of this